code/modules/recycling/sortingmachinery.dm

```diff
@@ -108,17 +158,21 @@
 	var/giftwrapped = 0
 	var/sortTag = 0
 	var/obj/item/paper/note
+	var/obj/item/barcode/sticker
 
 /obj/item/smallDelivery/contents_explosion(severity, target)
 	for(var/atom/movable/AM in contents)
 		AM.ex_act()
 
 /obj/item/smallDelivery/attack_self(mob/user)
 	user.temporarilyRemoveItemFromInventory(src, TRUE)
+	for(var/obj/I in src.GetAllContents())
+		SEND_SIGNAL(I, COMSIG_ITEM_UNWRAPPED)

```

Comment:
You have too many places you're using this signal and it looks like you need a more centralized place to handle unwrapping